corehq/apps/sso/models.py

```diff
@@ -404,6 +409,57 @@ def get_required_identity_provider(cls, username):
             return idp
         return None
 
+    def get_all_members_of_the_idp(self):
+        import msal
+        authority_base_url = "https://login.microsoftonline.com/"
+        authority = f"{authority_base_url}{self.api_host}"
+
+        config = {
+            "authority": authority,
+            "client_id": self.api_id,
+            "scope": ["https://graph.microsoft.com/.default"],
+            "secret": self.api_secret,
+            "endpoint": "https://graph.microsoft.com/v1.0/users"
+        }
+
+        try:
+            # Create a preferably long-lived app instance which maintains a token cache.
+            app = msal.ConfidentialClientApplication(
+                config["client_id"], authority=config["authority"],
+                client_credential=config["secret"],
+            )
+            # looks up a token from cache
+            result = app.acquire_token_silent(config["scope"], account=None)
+            if not result:
+                result = app.acquire_token_for_client(scopes=config["scope"])
+            if "access_token" in result:
+                # Calling graph using the access token
+                response = requests.get(
+                    config["endpoint"],
+                    headers={'Authorization': 'Bearer ' + result['access_token']},
+                )
+                response.raise_for_status()  # Raises an error for bad status
+                graph_data = response.json()
+
+                user_principal_names = [user["userPrincipalName"] for user in graph_data["value"]]
+                return user_principal_names
+            else:
+                error_message = f"Error acquiring token: {result.get('error')}, Description: "
+                f"{result.get('error_description')}, Correlation ID: {result.get('correlation_id')}"
+                raise Exception(error_message)
+        except Exception as e:
+            notify_exception(None, f"Failed to get members of the IdP. {str(e)}")
+
+            # Send email
+            subject = _("Issue Connecting to Microsoft Graph API")
+            body = f'There was an issue connecting to the Microsoft Graph API: {str(e)}'
```

Comment:
we should have a similar template [like the email sent here](https://github.com/dimagi/commcare-hq/blob/d58c94383e9883ba67c7e6355e9cd213f14617a5/corehq/apps/sso/utils/context_helpers.py#L51-L55)

It should read something like...
```
Dear enterprise administrator,

There was an issue connecting to the Microsoft Graph API when running a task to deactivate SSO Web Users on CommCare HQ that have been removed from your Microsoft Entra ID Enterprise Application.

This was the error message:
{{ error }}

Please check the configuration of the Remote User Management settings in your Identity Provider.

If you have any questions, please don’t hesitate to contact {{ contact_email }}.
Thank you for your use and support of CommCare.

Best regards,

The CommCare HQ Team
```

Reply:
Additionally, the send email portion of this task should probably be its own utility. likewise with the similar suggestion below